test/Microsoft.Health.Fhir.R4.Tests.Integration/Features/ChangeFeed/SqlServerFhirResourceChangeCaptureTests.cs

```diff
@@ -0,0 +1,313 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Messages.Delete;
+using Microsoft.Health.Fhir.SqlServer.Features.ChangeFeed;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Integration.Persistence;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Configs;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.Integration.Features.ChangeFeed
+{
+    /// <summary>
+    /// Integration tests for a resource change capture feature.
+    /// </summary>
+    public class SqlServerFhirResourceChangeCaptureTests
+    {
+        private const string LocalConnectionString = "server=(local);Integrated Security=true";
+        private IOptions<CoreFeatureConfiguration> _coreFeatureConfigOptions;
+
+        public SqlServerFhirResourceChangeCaptureTests()
+        {
+            _coreFeatureConfigOptions = Options.Create(new CoreFeatureConfiguration() { SupportsResourceChangeCapture = true });
+        }
+
+        /// <summary>
+        /// A basic smoke test verifying that the code can
+        /// insert and read resource changes after inserting a resource.
+        /// </summary>
+        [Fact]
+        public async Task GivenADatabaseSupportsResourceChangeCapture_WhenInsertingAResource_ThenResourceChangesShouldBeReturned()
+        {
+            FhirStorageTestsFixture fhirStorageTestsFixture = null;
+
+            try
+            {
+                for (int i = SchemaVersionConstants.SupportsResourceChangeCaptureSchemaVersion; i <= SchemaVersionConstants.Max; i++)
```

Comment:
Just so I understand, is there a reason why we need to test future schema versions?

Reply:
Wanted to validate all schema versions which support a resource change capture feature.

Reply:
I'm sorry to be a pain, but I am still not convinced this is necessary! So I can understand, may you outline a scenario where this would catch a backwards compatibility error?

I don't think the `FhirStorageTestsFixture` was intended to be instantiated multiple times, and I believe there could be a cost to testing with multiple schemas instead of using the test fixture that is already created for other integration tests.

Reply:
No worries. There are the following reasons why using the different instances of FhirStorageTestsFixture instead. 
- Validate the functionality for old schema versions.
- Need to initialize the SqlServerFhirDataStore with different settings. 
- Avoid breaking the existing test cases, and minimize the changes.

Reply:
I still have a few questions about this - would be great to chat and clear things up before you merge

Reply:
As discussed, added a fixture class to share a single database instance among all resource change capture tests.